index.js

```diff
@@ -265,57 +266,69 @@ RingPop.prototype.bootstrap = function bootstrap(opts, callback) {
         return;
     }
 
-    var start = new Date();
-
-    this.seedBootstrapHosts(bootstrapFile);
+    if (this.channel && this.channel.address() === null) {
+        this.channel.once('listening', doRealBootstrap);
```

Comment:
tchannel is now async open.

You cannot create it and send a request to it.

You have to wait for listening.

This makes bootstrap even more async.